parsing/extensions.mli

```diff
@@ -76,72 +79,28 @@ module Immutable_arrays : sig
   val pat_of : loc:Location.t -> pattern -> Parsetree.pattern_desc
 end
 
-(** The module type of language extension ASTs, instantiated once for each
-    syntactic category.  We tend to call the pattern-matching functions here
-    with unusual indentation, not indenting the [None] branch further so as to
-    avoid merge conflicts with upstream. *)
-module type AST = sig
-  (** The AST for all our ocaml-jst language extensions; one constructor per
-      language extension that extends the expression language.  Some extensions
-      are handled separately and thus are not listed here. *)
-  type t
-
-  (** The corresponding OCaml AST *)
-  type ast
-
-  (** Given an OCaml AST node, check to see if it corresponds to a language
-      extension term.  If it is, and the extension is enabled, then return it;
-      if it's not a language extension term, return [None]; if it's a disabled
-      language extension term, raise an error.
-
-      AN IMPORTANT NOTE: We indent calls to this function *very* strangely: we
-      *do not change the indentation level* when we match on its result!
-      E.g. from [type_expect_] in [typecore.ml]:
-
-      {[
-        match Extensions.Expression.of_ast sexp with
-        | Some eexp ->
-            type_expect_extension
-              ~loc ~env ~expected_mode ~ty_expected ~explanation eexp
-        | None      -> match sexp.pexp_desc with
-        | Pexp_ident lid ->
-            let path, mode, desc, kind = type_ident env ~recarg lid in
-            (* ... *)
-        | Pexp_constant(Pconst_string (str, _, _) as cst) ->
-            register_allocation expected_mode;
-            (* ... *)
-        | (* ... *)
-        | Pexp_unreachable ->
-            re { exp_desc = Texp_unreachable;
-                 exp_loc = loc; exp_extra = [];
-                 exp_type = instance ty_expected;
-                 exp_mode = expected_mode.mode;
-                 exp_attributes = sexp.pexp_attributes;
-                 exp_env = env }
-      ]}
-
-      Note that we match on the result of this function, forward to
-      [type_expect_extension] if we get something, and otherwise do the real
-      match on [sexp.pexp_desc] *without going up an indentation level*.  This
-      is important to reduce the number of merge conflicts with upstream by
-      avoiding changing the body of every single important function in the type
-      checker to add pointless indentation. *)
```

Comment:
I think we lost the comments about indentation, and they're very important documentation.  Could this comment or something like it move on to `get_desc`?